src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,111 @@
+package duke;
+import duke.task.TaskList;
+import duke.task.Task;
+import duke.task.Deadline;
+import duke.task.Todo;
+import duke.task.Event;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+public class Storage {
+    private static String dukeFilePath;
+
+    /**
+     * Check whether duke.txt exists and read lines from that file.
+     * @param taskList TaskList that manages tasks.
+     */
+    public static void loadFromFile(TaskList taskList) {
+        try {
+            Path currentRelativePath = Paths.get("");
+            String s = currentRelativePath.toAbsolutePath().toString();
+
+            java.nio.file.Path path = java.nio.file.Paths.get(s, "src", "data");
+            boolean directoryExists = java.nio.file.Files.exists(path);
+
+            if(!directoryExists) {
+                System.out.println("data folder does not exist. Let's create one.");
+                File dir = new File(String.valueOf(path));
+                dir.mkdir();
+            }
+            java.nio.file.Path filePath = java.nio.file.Paths.get(s, "src", "data", "duke.txt");
+            dukeFilePath = String.valueOf(filePath);
+            File f = new File(String.valueOf(filePath));
+            if (!f.createNewFile()) {
+                Scanner sc = new Scanner(f);
+                while (sc.hasNextLine()) {
+                    String row = sc.nextLine();
+                    String[] data = row.split(" {2}");
+//                  System.out.println(Arrays.toString(data));
+                    memoryProcessor(data, taskList);
+                }
+                sc.close();
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("Create new file please");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Adds the tasks from data to taskList.
+     * @param data Task info in data array.
+     * @param taskList TaskList that manages tasks.
+     */
+    public static void memoryProcessor(String[] data, TaskList taskList) {
+        switch (data[0]) {
+            case "T":
```

Comment:
For the `case` clauses, there should not be any indentation as well! It should be on the same level as `switch`